docs/guides.md

```diff
@@ -523,3 +523,307 @@ Now we enable autostart at the virtual machine level.
+```
+
+If the drive is added to the RAID array correctly, we should start to see a lot of disk activity and we should be able to monitor the progress of it by looking at the /proc/mdstat file.
+
+If the drive will not add to the array due to something left over on the drive, we should get an error from mdadm indicating the problem.  In that case we should be able to use the dd command to wipe out the one drive as above and then attempt to add it into the array.
+
+The other possibility is that the RAID array is created correctly but XCP-ng will not create a storage repository on it because some previous content of the drives is causing a problem.  It should be possible to recover from this by writing zeroes to the entire array without needing to rebuild it like this:
+
+```
+dd if=/dev/zero of=/dev/md0 bs=1M
+```
+
+After the probably very lengthy process of zeroing out the array, it should be possible to try again to create a storage repository on the RAID array.
+
+Another common cause for problems is a problem with either the mdadm.conf or dracut_mdraid.conf files.  Often when there is a problem with one of those files, the system will boot but fail to assemble or start the RAID arrays.  The boot RAID array will usually be found and assembled automatically but other RAID arrays may not.
+
+The best thing to do in this case is to check over the contents of the mdadm.conf and dracut_mdraid.conf files.  Look for mistyped or missing quotes in the dracut_mdraid.conf or missing spaces inside the quotes for those lines that have them.  Look for incorrect or mistyped UUID identifiers for the RAID arrays in both files.  The UUID identifiers should match the identifiers you get using the mdadm --examine --scan command and also match between the mdadm.conf file and the dracut_mdraid.conf file.  If any errors are found and corrected, rebuild the initrd file using the dracut command.
+
+In an extreme case, it should even be possible to delete and re-create those files using the normal instructions and rebuild the initrd file again.  It should also be possible but slightly more risky to remove the files and re-create the initrd file then reboot and attempt to re-create the files and initrd again after rebooting.
+
+Another possible but rare problem is caused by drives that shift their identifications from one system boot to the next.  A drive that has one name such as sdf on one boot might be different such as sdc on a different boot.  This is usually due to problems with the system BIOS or drivers and can also be caused by some hardware problems such as a drive taking wildly different amounts of time to start up from one boot to the next.  It it also more common with some types of storage such as NVMe storage.
+
+This type of problem is very difficult to diagnose and correct.  It may be possible to resolve it using different BIOS or setup configurations in the host system or by updating BIOS or controller firmware.
+
+### More and Different
```

Comment:
1. I think it would be good to mention the existence of this section at the beginning of your guide, so that people don't start by applying it step by step only to realize they should have done it differently.

2. Under this section shouldn't we cover the case of people who only want soft raid for their local SR but can't afford a disk for the system itself? (It can be a valid choice to favour redundancy of data over redundancy of the system)

Reply:
You're right about both.  I'll add those in.